lib/ignore/ignore.go

```diff
@@ -281,6 +289,22 @@ func (m *Matcher) ShouldIgnore(filename string) bool {
 	return false
 }
 
+// isInternal returns true if the file, as a path relative to the folder
+// root, represents an internal file that should always be ignored. The file
+// path must be clean (i.e., in canonical shortest form).
+func (m *Matcher) isInternal(file string) bool {
```

Comment:
> This relocation will cause me issues with the encrypted folder stuff.

To solve this, couldn't this code be moved into `Match` and `Result` be extended with a member function `IsInternal()` and accordingly an additional state `resultInternal`? As far as I understand your approach was to "interface" `Match` into the filesystem to first decrypt the filename, which would then still work.